src/typechecker.erl

```diff
@@ -5132,3 +5130,45 @@ number_of_exported_functions(Forms) ->
 
 line_no(Expr) ->
     erl_anno:line(element(2, Expr)).
+
+type_error(Kind) ->
+    {type_error, Kind}.
+
```

Comment:
What do these solve?

Reply:
The're just a bunch of functions without a spec. Isn't it better to define a type for all the errors?

Reply:
That's the next step perhaps? Anyway, this looks good so feel free to merge. :-)

Reply:
An `error` type won't hurt, indeed. I'll make it a bit more explicit in the next free time slot.